tools-v2/pkg/cli/command/curvebs/delete/snapshot/snapshot.go

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright (c) 2023 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveCli
+ * Created Date: 2023-11-10
+ * Author: ZackSoul
+ */
+package snapshot
+
+import (
+	"time"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	snapshotutil "github.com/opencurve/curve/tools-v2/internal/utils/snapshot"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	listSnapshot "github.com/opencurve/curve/tools-v2/pkg/cli/command/curvebs/list/snapshot"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/spf13/cobra"
+)
+
+const (
+	snapshotExample = `$ curve bs delete snapshot`
+)
+
+type SnapShotCommand struct {
+	basecmd.FinalCurveCmd
+	snapshotAddrs []string
+	timeout       time.Duration
+
+	user   string
+	file   string
+	uuid   string
+	failed bool
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*SnapShotCommand)(nil)
+
+func NewSnapShotCommand() *cobra.Command {
+	return NewDeleteSnapShotCommand().Cmd
+}
+
+func NewDeleteSnapShotCommand() *SnapShotCommand {
+	snapShotCommand := &SnapShotCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "snapshot",
+			Short:   "delete snapshot in curvebs",
+			Example: snapshotExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&snapShotCommand.FinalCurveCmd, snapShotCommand)
+	return snapShotCommand
+}
+
+func (sCmd *SnapShotCommand) AddFlags() {
+	config.AddBsSnapshotCloneFlagOption(sCmd.Cmd)
+	config.AddHttpTimeoutFlag(sCmd.Cmd)
+	config.AddBsUserOptionFlag(sCmd.Cmd)
+	config.AddBsSnapshotIDOptionFlag(sCmd.Cmd)
+	config.AddBsPathOptionFlag(sCmd.Cmd)
+	config.AddBsSnapshotFailedDOptionFlag(sCmd.Cmd)
+}
+
+func (sCmd *SnapShotCommand) Init(cmd *cobra.Command, args []string) error {
+	snapshotAddrs, err := config.GetBsSnapshotAddrSlice(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS || len(snapshotAddrs) == 0 {
```

Comment:
The two cases are diff, maybe they are not suitable for processing in a statement.